### PR TITLE
[W5.7][W12-3]Koh Jun Wei

### PR DESCRIPTION
**Pull request by JunWei96**
Delete command now accepts multiple indexes to delete multiple people.

### CODE REVIEW

File: test/java/seedu/addressbook/commands/DeleteCommandTest.java [hunk 1]
```diff
@@ -53,49 +55,61 @@ public void setUp() throws Exception {
...
-        assertDeletionFailsDueToInvalidIndex(1, addressBook, emptyDisplayList);
+        assertDeletionFailsDueToInvalidIndex(addressBook, emptyDisplayList, 1);
+        assertDeletionFailsDueToInvalidIndex(addressBook, emptyDisplayList, new int[] {400, 1, 23});
     }
 
     @Test
     public void execute_targetPersonNotInAddressBook_returnsPersonNotFoundMessage()
             throws IllegalValueException {
-        Person notInAddressBookPerson = new Person(new Name("Not In Book"), new Phone("63331444", false),
+        Person notInAddressBookPerson1 = new Person(new Name("Not In Booka"), new Phone("63331444", false),
```
Comment (hozongsien) [inline test/java/seedu/addressbook/commands/DeleteCommandTest.java:71]: Maybe can use the TestUtil class to generate a dummy person.

File: test/java/seedu/addressbook/commands/DeleteCommandTest.java [hunk 2]
```diff
@@ -145,18 +165,22 @@ private void assertDeletionFailsDueToNoSuchPerson(int visibleIndex, AddressBook
      *
      * @throws PersonNotFoundException if the selected person is not in the address book
      */
-    private void assertDeletionSuccessful(int targetVisibleIndex, AddressBook addressBook,
-                                          List<ReadOnlyPerson> displayList) throws PersonNotFoundException {
-
-        ReadOnlyPerson targetPerson = displayList.get(targetVisibleIndex - TextUi.DISPLAYED_INDEX_OFFSET);
-
+    private void assertMultipleDeletionSuccessful(AddressBook addressBook, List<ReadOnlyPerson> displayList,
```
Comment (hozongsien) [inline test/java/seedu/addressbook/commands/DeleteCommandTest.java:168]: What happens if one of the 2 ppl i want to delete is not found?

**Review by hozongsien — COMMENTED** [review]
Great job! Just some questions

File: test/java/seedu/addressbook/commands/DeleteCommandTest.java (hunk 1 above)
Comment (JunWei96) [reply test/java/seedu/addressbook/commands/DeleteCommandTest.java:71]: Yes I could, thanks for the feedback!

File: test/java/seedu/addressbook/commands/DeleteCommandTest.java (hunk 2 above)
Comment (JunWei96) [reply test/java/seedu/addressbook/commands/DeleteCommandTest.java:168]: Do you mean not found in the AddressBook? There are test cases for them. The test case is called 'assertDeletionFailsDueToNoSuchPerson'.
For every person that is not found, it will print out a line stating that the person is not found in the AddressBook.

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 3]
```diff
@@ -35,7 +35,19 @@ public void elementsAreUnique() throws Exception {
...
+    public void isAnyNull() throws Exception {
+        // empty list
+        assertFalse(Utils.isAnyNull());
+        
+        // null object exist
+        assertTrue(Utils.isAnyNull(null, "abc"));
+        
+        // null object does not exist
+        assertFalse(Utils.isAnyNull("abc", "def"));
+    }
```
Comment (damithc) [inline test/java/seedu/addressbook/common/UtilsTest.java:49]: In general, a PR should not contain unmerged code from other PRs. Try to limit each PR to one enhancement, which means the PR should be based on a branch that starts off from the `master` branch that is in sync with the upstream `master` branch

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -107,6 +107,10 @@ Examples:
 `delete 2` +
 Deletes the 2nd person in the address book.
 
+* `list` +
+`delete 1 2` +
+Deletes the 1st and 2nd person in the address book.
+
```
Comment (damithc) [inline docs/UserGuide.adoc:113]: Good job updating docs

File: src/seedu/addressbook/commands/Command.java [hunk 5]
```diff
@@ -14,13 +17,17 @@
 public class Command {
     protected AddressBook addressBook;
     protected List<? extends ReadOnlyPerson> relevantPersons;
-    private int targetIndex = -1;
+//    private int[] targetIndexes;
```
Comment (damithc) [inline src/seedu/addressbook/commands/Command.java:20]: Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

File: src/seedu/addressbook/commands/Command.java [hunk 6]
```diff
@@ -14,13 +17,17 @@
...
+    private Set<Integer> targetIndexes;
 
     /**
-     * @param targetIndex last visible listing index of the target person
+     * @param _targetIndexes last visible listing indexes of the target persons
      */
-    public Command(int targetIndex) {
-        this.setTargetIndex(targetIndex);
+    public Command(int... _targetIndexes) {
```
Comment (damithc) [inline src/seedu/addressbook/commands/Command.java:26]: This parameter name is not consistent with the rest of the codebase. i.e., there are no other cases of prefixing param name with an underscore

File: src/seedu/addressbook/parser/Parser.java [hunk 7]
```diff
@@ -32,7 +29,7 @@
     public static final Pattern PERSON_INDEX_ARGS_FORMAT = Pattern.compile("(?<targetIndex>.+)");
 
     public static final Pattern KEYWORDS_ARGS_FORMAT =
-            Pattern.compile("(?<keywords>\\S+(?:\\s+\\S+)*)"); // one or more keywords separated by whitespace
+                Pattern.compile("(?<keywords>\\S+(?:\\s+\\S+)*)"); // one or more keywords separated by whitespace
```
Comment (damithc) [inline src/seedu/addressbook/parser/Parser.java:32]: Wrong indentation? [Our coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#layout) requires 4 spaces (instead of tabs) for normal indentation and an indentation of 8 spaces for line wrapping (i.e. when breaking a long line into multiple lines).

Comment (damithc) [reply src/seedu/addressbook/parser/Parser.java:32]: Furthermore, this change is not related to the enhancement proposed by this PR

File: src/seedu/addressbook/ui/Formatter.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+    /**
+     * Formats a string as a viewable indexed list item.
+     *
+     * @param visibleIndex visible index for this listing
+     */
+    private static String getIndexedListItem(int visibleIndex, String listItem) {
+        return String.format(MESSAGE_INDEXED_LIST_ITEM, visibleIndex, listItem);
+    }
+}
```
Comment (damithc) [inline src/seedu/addressbook/ui/Formatter.java:79]: This change does not seem to be related to the enhancement proposed by this PR :thinking:

**Review by damithc — APPROVED** [review]
:+1: Good effort on this PR @JunWei96  In particular, good job on
* updating docs
* updating I/O tests
* updating JUnit tests

Suggestions to improve further:
* Limit the PR to one enhancement and avoid including changes not related to that enhancement or unmerged  code from other PRs

**Comment by damithc** [comment]
As before, please close the PR after viewing comments.
